File: app/src/main/java/com/kota/Bahamut/Pages/Model/MailBoxPageBlock.java
package com.kota.Bahamut.Pages.Model;

import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import java.util.Stack;

public class MailBoxPageBlock extends TelnetListPageBlock {
    private static Stack<MailBoxPageBlock> _pool = new Stack<>();

    private MailBoxPageBlock() {
    }

    public static void release() {
        synchronized (_pool) {
            _pool.clear();
        }
    }

    public static void recycle(MailBoxPageBlock block) {
        synchronized (_pool) {
            _pool.push(block);
        }
    }

    public static MailBoxPageBlock create() {
        MailBoxPageBlock block = null;
        synchronized (_pool) {
            if (_pool.size() > 0) {
                block = _pool.pop();
            }
        }
        if (block == null) {
            return new MailBoxPageBlock();
        }
        return block;
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/model/MailBoxPageBlock.kt
package com.kota.Bahamut.pages.model

import com.kota.Bahamut.listPage.TelnetListPageBlock
import java.util.Stack

class MailBoxPageBlock : TelnetListPageBlock() {
    
    companion object {
        private val myPool = Stack<MailBoxPageBlock?>()

        @JvmStatic
        fun release() {
            synchronized(myPool) {
                myPool.clear()
            }
        }

        @JvmStatic
        fun recycle(block: MailBoxPageBlock?) {
            synchronized(myPool) {
                myPool.push(block)
            }
        }

        fun create(): MailBoxPageBlock {
            var block: MailBoxPageBlock? = null
            synchronized(myPool) {
                if (myPool.isNotEmpty()) {
                    block = myPool.pop()
                }
            }
            if (block == null) {
                return MailBoxPageBlock()
            }
            return block
        }
    }
}
